### PR TITLE
moved some fields from PeerManagerActor to NetworkState

### PR DESCRIPTION
**Pull request by pompon0**
...so that PeerActor has a direct access to them.
In particular RoutingTableView has been adjusted for concurrent access.
Also event_sink (test-only) has been moved even deeper (to NetworkConfig).

### CODE REVIEW

File: chain/network/src/routing/routing_table_view.rs [hunk 1]
```diff
@@ -54,159 +106,126 @@ impl RoutingTableView {
...
         let account_id = announce_account.account_id.clone();
-        self.account_peers.put(account_id.clone(), announce_account.clone());
-
+        inner.account_peers.put(account_id.clone(), announce_account.clone());
         // Add account to store
-        if let Err(e) = self.store.set_account_announcement(&account_id, &announce_account) {
+        if let Err(e) = inner.store.set_account_announcement(&account_id, &announce_account) {
             warn!(target: "network", "Error saving announce account to store: {:?}", e);
         }
     }
```
Comment (matklad) [inline chain/network/src/routing/routing_table_view.rs:166]: Not super happy about the current state. Here we are locking a global data-structure while accessing the storage. This is fine as an intermediate step, but I wish long-term to make sure that long-term we minimize the scope of locks. This type is basically "lock whenever I do anything", which is error-prone.

**Review by matklad — APPROVED** [review]
Broadly LGTM, but RoutingTableView could perhaps beenfit from replacing low-level manipulation APIs with high-level APIs which take the lock only once. My primary concern here is not perforamnce, but transactional semantics of updates.

File: chain/network/src/peer_manager/peer_manager_actor.rs [hunk 2]
```diff
@@ -158,6 +157,17 @@ pub(crate) struct NetworkState {
     pub accounts_data: Arc<accounts_data::Cache>,
     /// Connected peers (inbound and outbound) with their full peer information.
     pub connected_peers: ConnectedPeers,
+
```
[the remark this answers is not in the document]
Comment (pompon0) [reply chain/network/src/peer_manager/peer_manager_actor.rs:160]: SG, but in a separate PR.

File: chain/network/src/routing/routing_table_view.rs (hunk 1 above)
Comment (matklad) [reply chain/network/src/routing/routing_table_view.rs:166]: > It was already under a global mutex of PeerManagerActor, and now it gets its own global mutex

Not quite: while in PMA, we had transactional semantics: we removed *a bunch* of accounts ate the same time, in lockstep with changing next hops, etc. Now it is possible to observe intermediate states.

Comment (pompon0) [reply chain/network/src/routing/routing_table_view.rs:166]: ok, will fix that

File: chain/network/src/config.rs [hunk 3]
```diff
@@ -276,6 +280,7 @@ impl NetworkConfig {
             accounts_data_broadcast_rate_limit: demux::RateLimit { qps: 100., burst: 1000000 },
             features: Features { enable_tier1: true },
             skip_sending_tombstones: None,
+            event_sink: Sink::void(),
```
[the remark this answers is not in the document]
Comment (pompon0) [reply chain/network/src/config.rs:283]: done

File: chain/network/src/routing/routing_table_view.rs (hunk 1 above)
Comment (pompon0) [reply chain/network/src/routing/routing_table_view.rs]: done, PTAL

File: chain/network/src/routing/routing_table_view.rs [hunk 4]
```diff
@@ -54,159 +106,126 @@ impl RoutingTableView {
...
                 return;
             }
-            self.local_edges_info.insert(other_peer.clone(), edge);
+            inner.local_edges_info.insert(other_peer.clone(), edge);
         }
     }
 
-    pub(crate) fn remove_local_edge(&mut self, peer_id: &PeerId) {
-        self.local_edges_info.remove(peer_id);
+    pub(crate) fn remove_local_edge(&self, peer_id: &PeerId) {
```
[the remark this answers is not in the document]
Comment (pompon0) [reply chain/network/src/routing/routing_table_view.rs]: done

File: chain/network/src/peer_manager/peer_manager_actor.rs [hunk 5]
```diff
@@ -448,20 +466,20 @@ impl PeerManagerActor {
...
-        accounts.retain(|a| !self.routing_table_view.contains_account(&a));
+        accounts.retain(|a| !self.state.routing_table_view.contains_account(&a));
         if accounts.is_empty() {
             return;
         }
         debug!(target: "network", account_id = ?self.config.validator.as_ref().map(|v|v.account_id()), ?accounts, "Received new accounts");
         for account in &accounts {
-            self.routing_table_view.add_account(account.clone());
+            self.state.routing_table_view.add_account(account.clone());
         }
```
[the remark this answers is not in the document]
Comment (pompon0) [reply chain/network/src/peer_manager/peer_manager_actor.rs]: done

File: chain/network/src/config.rs [hunk 6]
```diff
@@ -122,13 +124,14 @@ pub struct NetworkConfig {
 
     // Whether to send tombstones at startup.
     pub skip_sending_tombstones: Option<time::Duration>,
+
+    /// TEST-ONLY
+    /// TODO(gprusak): make it pub(crate), once all integration tests
+    /// are merged into near_network.
+    pub event_sink: Sink<Event>,
```
[the remark this answers is not in the document]
Comment (pompon0) [reply chain/network/src/config.rs:131]: in fact, I'm planning to pass the TCP listen port via config as well. This is kind of similar :P.

[the remark this answers is not in the document]
Comment (matklad) [reply chain/network/src/config.rs:131]: s/Config/Context/ would fix that :)

**Review by matklad — APPROVED** [review]

**Pull request merged**